### PR TITLE
Setup CI

### PR DESCRIPTION
**Pull request by chapulina**
Add GitHub actions-based CI that runs on every push and PR.

It uses binaries for ROS and Gazebo dependencies, pulled from https://packages.ros.org, and compiles from source all the repos in [buoy_all.yaml](https://github.com/osrf/buoy_entrypoint/blob/main/buoy_all.yaml). 

Then it tests all buoy repos, including `buoy_msgs`. I think we should just rely on the ROS buildfarm to run CI for the messages package eventually, but until we're ready to bloom it, we can test it here.

CI is currently failing the copyright check, which should be fixed by #6

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  tests:
+    name: Build and test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - gz-version: "fortress"
+            ros-distro: "galactic"
```
Comment (quarkytale) [inline .github/workflows/ci.yml:13]: Is there supposed to be another hyphen? 
`- ros-distro: "galactic"`

File: .github/workflows/build-and-test.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -ev
```
Comment (quarkytale) [inline .github/workflows/build-and-test.sh:2]: Not sure about this, what's `ev`, or its `env`?

**Comment by quarkytale** [comment]
Had retriggered checks, but its only failing for copyright

File: .github/workflows/ci.yml (hunk 1 above)
Comment (chapulina) [reply .github/workflows/ci.yml:13]: I believe it's one hyphen per matrix. See the tutorial [here](https://docs.github.com/en/github-ae@latest/actions/using-jobs/using-a-build-matrix-for-your-jobs#example-including-new-combinations). For now we only support one combination, `fortress` + `galactic`, but in the future we may want to extend to other combos (i.e. `fortress` + `humble`), and that would go on a new hyphen.

File: .github/workflows/build-and-test.sh (hunk 2 above)
Comment (chapulina) [reply .github/workflows/build-and-test.sh:2]: This is actually the same as `set -e -v`. From `set --help`:

* _-e  Exit immediately if a command exits with a non-zero status._ - this is good for CI so we don't waste time trying other commands when one of them fails
* _-v  Print shell input lines as they are read._ - this is helpful when we're reading the build logs, to know what commands were running

**Review by quarkytale — APPROVED** [review]

**Comment by chapulina** [comment]
Thanks for the review, @quarkytale ! Now that #2 and  #5 have been merged, I expect the linters to fail.

Once CI comes back, if only the linters are failing, I'll merge this PR, and you can work against `main` to address those linters as part of #3 until CI is 🟢 . If something else fails, like the build, I'll address it in this PR before merging.

**Comment by chapulina** [comment]
Ok, only linter errors:

```
	  1 - copyright (Failed)
	  2 - cppcheck (Failed)
	  3 - cpplint (Failed)
```

Merging!

**Pull request merged**